packages/contracts/test/ts/unlimited_allowance_token_v2.ts

```diff
@@ -0,0 +1,129 @@
+        });
+
+        it('should return true on a 0 value transfer', async () => {
+            const amountToTransfer = 0;
+            const didReturnTrue = await token.transferFrom.call(owner, spender, amountToTransfer, {from: spender});
+            expect(didReturnTrue).to.be.true();
+        });
+
+        it('should not modify spender allowance if spender allowance is 2^256 - 1', async () => {
+            const initOwnerBalance = await zeroEx.token.getBalanceAsync(tokenAddress, owner);
+            const amountToTransfer = initOwnerBalance;
+            const initSpenderAllowance = zeroEx.token.UNLIMITED_ALLOWANCE_IN_BASE_UNITS;
+            await zeroEx.token.setAllowanceAsync(tokenAddress, owner, spender, initSpenderAllowance);
+            await zeroEx.token.transferFromAsync(tokenAddress, owner, spender, spender, amountToTransfer,
+                                                 {gasLimit: constants.MAX_TOKEN_TRANSFERFROM_GAS});
+
+            const newSpenderAllowance = await zeroEx.token.getAllowanceAsync(tokenAddress, owner, spender);
+            expect(initSpenderAllowance).to.be.bignumber.equal(newSpenderAllowance);
+        });
+
+        it('should transfer the correct balances if spender has sufficient allowance', async () => {
+            const initOwnerBalance = await zeroEx.token.getBalanceAsync(tokenAddress, owner);
+            const amountToTransfer = initOwnerBalance;
+            const initSpenderAllowance = initOwnerBalance;
+            await zeroEx.token.setAllowanceAsync(tokenAddress, owner, spender, initSpenderAllowance);
+            await zeroEx.token.transferFromAsync(tokenAddress, owner, spender, spender, amountToTransfer,
+                                                 {gasLimit: constants.MAX_TOKEN_TRANSFERFROM_GAS});
+
+            const newOwnerBalance = await zeroEx.token.getBalanceAsync(tokenAddress, owner);
+            const newSpenderBalance = await zeroEx.token.getBalanceAsync(tokenAddress, spender);
+
+            expect(newOwnerBalance).to.be.bignumber.equal(0);
+            expect(newSpenderBalance).to.be.bignumber.equal(initOwnerBalance);
+        });
+
+        it('should modify allowance if spender has sufficient allowance less than 2^256 - 1', async () => {
+            const initOwnerBalance = await zeroEx.token.getBalanceAsync(tokenAddress, owner);
+            const amountToTransfer = initOwnerBalance;
+            const initSpenderAllowance = initOwnerBalance;
+            await zeroEx.token.setAllowanceAsync(tokenAddress, owner, spender, initSpenderAllowance);
+            await zeroEx.token.transferFromAsync(tokenAddress, owner, spender, spender, amountToTransfer,
+                                                 {gasLimit: constants.MAX_TOKEN_TRANSFERFROM_GAS});
+
+            const newSpenderAllowance = await zeroEx.token.getAllowanceAsync(tokenAddress, owner, spender);
+            expect(newSpenderAllowance).to.be.bignumber.equal(0);
+        });
+    });
+});
```

Comment:
The test duplication is also painful.

Reply:
Yeah, we need to figure out the best structure for testing within contract packages as well.